controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -563,23 +563,25 @@ func clearHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) {
 }
 
 func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
-	// Adopt the host in case it has been re-registered during the
-	// deprovisioning process before it completed
-	provResult, err := prov.Adopt(info.host.Status.ErrorType == metal3v1alpha1.RegistrationError)
-	if err != nil {
-		return actionError{err}
-	}
-	if provResult.ErrorMessage != "" {
-		return recordActionFailure(info, metal3v1alpha1.RegistrationError, provResult.ErrorMessage)
-	}
-	if provResult.Dirty {
-		info.host.ClearError()
-		return actionContinue{provResult.RequeueAfter}
+	if info.host.Status.Provisioning.Image.URL != "" {
```

Comment:
This is going to have implications for the work @hardys is doing to add live ISO image support, since that PR adds a second URL field to check. I wonder if that means we should have gone with an image mode flag instead of separate fields.